### ISSUE TITLE
deleteAllWithError does not delete an entity.

### ISSUE THREAD

**Issue by sakshiGoel26** [chriscdn/RHManagedObject#26]
We are facing an issue in our project, sometimes using "deleteAllWithError" for an entity doesn't delete it.
Help will be appreciated.

**Comment by chriscdn** [comment]
Is it possible you're deleting the items in one thread and the items that persist are being managed and saved in another thread?

**Comment by sakshiGoel26** [comment]
@chriscdn I am saving & deleting items on main thread only. While deleting I am using "deleteAllWithError" method of RHMangedObject which internally uses "managedObjectContextForCurrentThreadWithError" method & saving process uses "managedObjectContextManager" method. 
Is this issue happening due to this?

**Comment by chriscdn** [comment]
> Is this issue happening due to this?

I don't think so.  In the end both methods use the `NSManagedObjectContext` for the current thread, which is the same if you're doing this all from the same thread.  Something else to check: Can you confirm the items not being deleted have actually been saved?

**Comment by sakshiGoel26** [comment]
Issue resolved. Reason was, after deleting an entity we were not using "commit" for that entity.
Thanks.
